components/authz-service/storage/postgres/migration/migration.go

```diff
@@ -121,18 +126,26 @@ func (c *Config) Migrate(dataMigConf datamigration.Config) error {
 	}
 
 	// idempotent
+	l.Info("Migrating data if needed...")
 	err = dataMigConf.Migrate()
+	l.Info("Migrating data if needed -- complete")
 	if err != nil {
 		return errors.Wrap(err, "IAM data migrations failed")
 	}
 
 	// perform remaining migrations
+	l.Infof("Migrating schema post-version %v...", PreForceUpgradeMigration)
 	err = m.Up()
 	if err != nil && err != migrate.ErrNoChange {
 		return errors.Wrap(err, "migrations failed")
 	}
+	resolution := "complete"
+	if err == migrate.ErrNoChange {
+		resolution = "no changes necessary"
+	}
+	l.Infof("Migrating schema post-version %v -- %s", PreForceUpgradeMigration, resolution)
```

Comment:
Notice the `resolution` on the end; it is either "complete" or "no changes necessary", which I think is very useful to share.
Could do the same thing for data migrations (L129-L131) above, but that requires changing what `dataMigConf.Migrate()` returns. Is that worth doing in your opinions?

Reply:
Will the `resolution` change once we add new migrations?

What would have to change about `dataMigConf.Migrate()`? Not sure I feel strongly either way. What does `complete` vs. `no changes necessary` tell us?

Reply:
I don't think we need the addition `resolution` in datamigrations. We can always add it if we need to.

Reply:
@bcmdarroch 
(1) Yes, as we add new migrations and the customer restarts authz, they will either see "complete" or "no changes necessary" as appropriate. 
(2) For dataMigConf, it would need to also return the `migrate.ErrNoChange` error so we could do the same check.

Reply:
Hmm, still thinking about the difference between `complete` vs. `no changes necessary`. By `complete` you mean to say that all migrations that should have been applied were actually applied, right? That might be better communicated by `latest migrations applied`.

Reply:
I really want the verb signifying, well, completion, to be identical for all items, so I really want it to be "complete". If you want to tweak the wording on the left side, we could do that, but that wording needs to be the same for the opening and closing "brackets".

Reply:
what if we replaced `changes` with `migrations`?

Reply:
also maybe instead of `complete` we use `migration complete`